plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -0,0 +1,147 @@
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.ml.common.MLModelGroup;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupAction;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupInput;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupRequest;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupResponse;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.ml.utils.SecurityUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import com.google.common.collect.ImmutableList;
+
+@Log4j2
+public class TransportUpdateModelGroupAction extends HandledTransportAction<ActionRequest, MLUpdateModelGroupResponse> {
+
+    private final TransportService transportService;
+    private final ActionFilters actionFilters;
+    private Client client;
+    private NamedXContentRegistry xContentRegistry;
+
+    @Inject
+    public TransportUpdateModelGroupAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(MLUpdateModelGroupAction.NAME, transportService, actionFilters, MLUpdateModelGroupRequest::new);
+        this.actionFilters = actionFilters;
+        this.transportService = transportService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLUpdateModelGroupResponse> listener) {
+        MLUpdateModelGroupRequest updateModelGroupRequest = MLUpdateModelGroupRequest.fromActionRequest(request);
+        MLUpdateModelGroupInput updateModelGroupInput = updateModelGroupRequest.getUpdateModelGroupInput();
+
+        String modelGroupId = updateModelGroupInput.getModelGroupID();
+        GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+
+        if (modelGroupId != null) {
+            User user = RestActionUtils.getUserContext(client);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                client.get(getModelGroupRequest, ActionListener.wrap(modelGroup -> {
+                    if (modelGroup.isExists()) {
+                        Map<String, Object> source = modelGroup.getSourceAsMap();
+                        Map<String, Object> owner = (Map) source.get(MLModelGroup.OWNER);
+                        String access = (String) source.get(MLModelGroup.ACCESS);
+
+                        if (SecurityUtils.isAdmin(user) || access == "public") {
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else if (user.getName().equals(owner.get("name"))) {
```

Comment:
Always add null check to user since it could be null if the security plugin is not enabled.

Reply:
Good catch. Thanks Zan, will add the check wherever needed